docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,495 @@
+
+    def on_incoming_connection_complete(self, anchor: namedtuple) -> None:
+        # truncated code
+
+    def on_record_batch(self, batch: "Table", anchor: namedtuple) -> None:
+        # truncated code
+
+    def on_complete(self) -> None:
+        import pandas as pd
+        import pyarrow as pa
+
+        df = pd.DataFrame(
+            {
+                "x": [1, 2, 3],
+                "y": ["hello", "world", "from ayx_python_sdk!"],
+                "z": [self.config_value, self.config_value, self.config_value],
+            }
+        )
+
+        packet = pa.Table.from_pandas(df)
+
+        self.provider.write_to_anchor("Output", packet)
+        self.provider.io.info("DanceableLyrics tool done.")
+```
+
+## Write a Plugin
+In this guide, we use multiple data files. We read each directly to illustrate a connector-like plugin. That is, to connect directly to local or remote data sources. While here, we read CSV input files, these could be in any format, including formats that Alteryx Designer does not directly support.
+
+> :information_source: To see an Input tool that connects to a remote API, check out [Creating an API Tool](https://github.com/alteryx/ayx-developer-sdk/blob/main/docs/howto/how-to-make-api-tool/api-tool-guide.md)!
+
+> :information_source: Since this is an Input tool, we only focus on the `__init__` and `on_complete` functions. For additional information on the lifecycle of a plugin, refer to the [AYX Python SDK documentation](https://alteryx.github.io/ayx-python-sdk/plugin_lifecycle.html).
+
+### 1. Dependencies
+The first step is to update our `./backend/requirements-thirdparty.txt` file and tell Python we depend on Polars as a 3rd-party dependency. To do this, add a single line to the file:
+
+```txt
+polars==0.17.9
+```
+
+### 2. Imports
+The next step is to update our Python module's imports. Imports tell Python what other code we will reference in our plugin. Add the required import statements near the top of your `danceable_lyrics.py` file. Your code should look like this:
+
+```python
+from collections import namedtuple
+
+from ayx_python_sdk.core import PluginV2
+from ayx_python_sdk.core.field import FieldType
+from ayx_python_sdk.core.utils import create_schema
+from ayx_python_sdk.providers.amp_provider.amp_provider_v2 import AMPProviderV2
+
+import polars as pl
+import pyarrow as pa
+```
+
+### 3. Initialization
+Now, let's update the plugin's `__init__` method and set up some basic variables that we will reference later in the code. To start, modify the start of `__init__` to set a base path to _where you extracted your data assets_. Use `self.DATASETS_BASE` to do so. The result should look something like this:
+
+```python
+def __init__(self, provider: AMPProviderV2) -> None:
+    """Construct a plugin."""
+    self.provider = provider
+    self.name = "DanceableLyrics"
+
+    # Point this variable to your extracted datasets directory:
+    self.DATASETS_BASE = "c:/users/alteryx/DanceableLyricsData/"
```

Comment:
Probably add a disclaimer that people need to set the path. People often forget to do this when they copy paste.

Reply:
This is covered in the documentation